File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int n = s.length();
        int l = 0, r = 0, max = 0;
        Map<Character, Integer> mpp = new HashMap<>();

        while (r < n) {

            if (mpp.containsKey(s.charAt(r))) {
                l = Math.max(l, mpp.get(s.charAt(r)) + 1);
            }
            mpp.put(s.charAt(r), r);
            max = Math.max(max, (r - l + 1));
            r++;
        }
        return max;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l=r=0
        n = len(s)
        arr =[]
        res =0
        while r<n:
            ch =s[r]

            while ch in arr:
                arr.remove(s[l])
                l+=1
            arr.append(s[r]) 
            res = max(res,r-l+1)

            r+=1
        return res        